src/runtime_src/core/tools/xbutil2/SubCmdValidate.cpp

```diff
@@ -1673,10 +1728,28 @@ SubCmdValidate::execute(const SubCmdOptions& _options) const
         xclbin_location.append("/");
     }
 
+    //check if param option is provided
+    if (!sParam.empty()) {
+      XBU::verbose("Sub command: --param");
+      boost::split(param, sParam, boost::is_any_of(":")); // eg: dma:block-size:1024
+
+      //check parameter format
+      if (param.size() != 3)
+        throw xrt_core::error((boost::format("Invalid parameter format : '%s'") % sParam).str());
+      
+      //check test case name
+      doesTestExists(param[0], testNameDescription);
```

Comment:
This should occur, before checking the parameter size.  And yes, we know that there is at lease one entry.